libr/asm/p/asm_x86_nz.c

```diff
@@ -1793,7 +1793,7 @@ static int opmov(RAsm *a, ut8 *data, const Opcode *op) {
 		if (!op->operands[1].is_good_flag) {
 			return -1;
 		}
-		if (op->operands[1].immediate == -1) {
+		if (op->operands[1].immediate == -1 && a->num && a->num->nc.errors > 0) {
```

Comment:
Will be good to not depend on RNum at all. So this conditional should work when a.num is null

Reply:
Fix this and ill merge

Reply:
I'm not that desperate to merge this one in but ... you have a deal though I need at least another 24h

Reply:
> Will be good to not depend on RNum at all

ok 22cf08b6f0cce706372a3932d8622071c0912d88 ... i haven't reverted the changes to `libr/util/unum.c` since they are improvements